### DISCUSSION TITLE
@live directive queries with a custom resolver

### DISCUSSION THREAD

**Discussion by chillichelli** [graphprotocol/graph-client#280] [Q&A]
Hi!

Im trying to use the @live directive in combination with a custom resolver but I cant seem to do so.

The query looks like this:
![Schermafbeelding 2022-10-10 om 14 46 23](https://user-images.githubusercontent.com/80068137/194869566-62c53bf4-4585-4e40-ae95-99aa3610f050.png)

Where it calls crossChainTransactions, our Query type is extended using `additionalTypeDefs` with this named query in our .graphclientrc.yml like so:
![Schermafbeelding 2022-10-10 om 14 47 52](https://user-images.githubusercontent.com/80068137/194869831-f66bad82-f3a9-45bc-8013-c82b757c3f07.png)

Now this is the part where im getting confused, im not really sure what to put in the resolver
![Schermafbeelding 2022-10-10 om 14 48 25](https://user-images.githubusercontent.com/80068137/194869931-58955351-6136-4d4f-8040-a3a1c9daaf92.png)

Thanks in advance,

**Comment by ardatan** [comment]
Not sure I follow. `@live` is an operation directive not field, so it has nothing to do with resolvers. It just specifies how you execute the operation.

**Reply by chillichelli** [reply]
Yes exactly, so now line 48 and 63 in this example return a AsyncIterable. Im not sure how I can define in the `additionalTypeDefs` in our `.graphclientrc.yml` that the function `crossChainTransactions` returns a AsyncIterable instead of a Pair
![Schermafbeelding 2022-10-10 om 17 41 55](https://user-images.githubusercontent.com/80068137/194904621-2ef1839a-fc50-42d4-852d-aef0f4276196.png)

Obviously im seeing this error now:
![Schermafbeelding 2022-10-10 om 17 46 07](https://user-images.githubusercontent.com/80068137/194905444-cbf4b323-ce0e-4f57-973f-a3850918898c.png)
